### ISSUE TITLE
Do not allow section anchor to wrap alone to next line

### ISSUE THREAD

**Issue by benjaoming** [readthedocs/sphinx_rtd_theme#1412]
We need a CSS rule that prohibits the Pilcrow (¶) from jumping to the next line. It's hidden so it looks weird.

![image](https://user-images.githubusercontent.com/374612/212976147-80c08e45-f3fa-406b-96f8-94009d8bf96d.png)

**Comment by n-peugnet** [comment]
This can be done by adding the following CSS (only tested on Firefox for now):

```css
.rst-content a.headerlink {
  display: inline;
}
```

As no space is present between the headerlink and the previous word, it forces them to be wrapped together.

![Peek 2023-01-20 15-15](https://user-images.githubusercontent.com/23519418/213720530-f17e6276-b74f-4a2b-87d7-d7d6280a8b87.gif)

**Comment by benjaoming** [comment]
> As no space is present between the headerlink and the previous word

Ah yeah, perhaps we can add a `&nbsp;` - I guess this is exactly what it is meant for :)

**Comment by n-peugnet** [comment]
> > As no space is present between the headerlink and the previous word
> 
> Ah yeah, perhaps we can add a `&nbsp;` - I guess this is exactly what it is meant for :)

Not really, the problem comes from the fact that currently the `.headerlink` is set to `display: inline-block`, so it will always first wrap at the block boundary. With no CSS at all, the behaviour would be the expected one, as the `<a>` is not separated from the previous word with any space. Using a `&nbsp` here will only add complexity, adding space with `margin-left` as currently seems a better idea.

I never used SASS but it seems `display: inline` (or `display: initial`) can be added here: https://github.com/readthedocs/sphinx_rtd_theme/blob/6444ca893043d8a0ca704474f7cf7d5a148f6767/src/sass/_theme_rst.sass#L216-L226

**Comment by benjaoming** [comment]
Roger! Thanks for the wonderful explanation!

If you would like to submit a PR, I can build the SASS and push to your PR :+1: Alternatively, there is a `Makefile` in our project with some Docker build stuff.

**Comment by n-peugnet** [comment]
> If you would like to submit a PR, I can build the SASS and push to your PR

I think that I prefer this option. I will make the PR right now.
